### DISCUSSION TITLE
What's reason/system design to not make a user as a content item like V.1?

### DISCUSSION THREAD

**Discussion by aaronamm** [OrchardCMS/OrchardCore#12000] [Q&A] [answered]
In Orchard  CMS (V.1) a user is a content item:
https://github.com/OrchardCMS/Orchard/blob/dev/src/Orchard.Web/Modules/Orchard.Users/Migrations.cs#L54
https://github.com/OrchardCMS/Orchard/blob/dev/src/Orchard.Web/Modules/Orchard.Users/Handlers/UserPartHandler.cs#L9

However in Orchard Core CMS (V.2), it has been changed to an Entity which is not a content item:
https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore/OrchardCore.Users.Core/Models/User.cs#L8

For curiosity, would it be possible to let me know a system design behind this implementation and a reason that not make a user as a content item?

I think making a user as a content type has some benefit e.g. we can use an existing content item infrastructure for a user. 

Thank you so much.

CC @Skrypt

**Comment by Skrypt** [comment, marked as answer]
A Content-Type is mutable from runtime which is not that great for security purposes.
There are examples of how to extend the current IUser interface for your own usage which I think makes more sense.
You are not the first one to ask this question. Take a look at this Pull Request for example  

https://github.com/OrchardCMS/OrchardCore/pull/3141

This PR extends the IUser with an IProfile entity that can be used for FrontEnd users.
It is not even extending the IUser directly. It just adds an IProfile that links to the IUser.

Now, if you want to extend an IUser from runtime you can also use the UserContentPicker on a Content-Type to link them together.
You can also create a driver that will extend the current User editing admin UI.
There is an example of that in OrchardCore.Demo.

`public class UserProfileDisplayDriver : SectionDisplayDriver<User, UserProfile>`

**Reply by jtkech** [reply]
There is also the `OrchardCore.Users.CustomUserSettings` feature allowing content types to become custom user settings, it's like adding fields to the user, I'm using it, there is a demo video on youtube.

https://github.com/OrchardCMS/OrchardCore/blob/7ecec9da3f18134f3cb0a7a82fa5014bf3bba653/src/OrchardCore.Modules/OrchardCore.Users/Manifest.cs#L50

**Reply by aaronamm** [reply]
@Skrypt thank you so much for your prompt reply and explained why we don't make a user as a content type/content item.

**Reply by aaronamm** [reply]
@jtkech  thank you for additional information.  I would like to try the first approach that `Márk Bartha` suggested in Orchard Core Gitter.
Here are two approached that he suggested:

To add more fields to the registration page (as well as login page) you need to implement something similar that you can see in the ReCaptcha module. You can find an MVC filter which injects a shape to "AfterRegister" zone if the displayed page is the register page. To validate it you need to create a IRegisterFormEvents implementation. Filter: https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.ReCaptcha/ReCaptchaLoginFilter.cs and FormEvents: https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.ReCaptcha/Users/Handlers/RegistrationFormHandler.cs
Note that the User is not a Content Item so you need to handle the result accordingly like creating a Content Item in the RegisteredAsync event of your event handler.

Another completely different solution is to create a custom form for the registration page and a workflow that handles it (i.e. creates an Orchard Core user). Here you can add your fields but still you need to manage creating the Content Item from the workflows. See this documentation page: https://docs.orchardcore.net/en/dev/docs/reference/modules/Forms/
This might be possible without coding.

This is useful for me as well.
I don't want to override AccountController and looking for a better approach.

**Reply by hishamco** [reply]
This is the [video](https://github.com/OrchardCMS/OrchardCore/tree/532222e1a4d492281ca9dd1990f237d90e8ed382/src/docs/reference/modules/CustomSettings#video) that @jtkech refers to

**Reply by aaronamm** [reply]
@hishamco thank you for the link the video.
